syz-verifier/main.go

```diff
@@ -251,14 +262,9 @@ func (srv *RPCServer) NextExchange(a *rpctype.NextExchangeArgs, r *rpctype.NextE
 			Info:   a.Info,
 		}
 		if srv.newResult(res, a.ProgIdx) {
-			for idx, prog := range srv.progs {
-				if a.ProgIdx == prog.idx {
-					if !verf.Verify(prog.res, prog.prog) {
-						log.Printf("mismatch found for %+v", prog.prog)
-					}
-					delete(srv.progs, idx)
-				}
-			}
+			prog := srv.progs[a.ProgIdx]
```

Comment:
newResult maps prodIdx to progInfo, and we do the same a line below.
If you map progIdx to progInfo as early as possible once, and then pass it around, it will simply and shorten the code. E.g. newResult can then be just:
```
func (srv *RPCServer) newResult(prog *progInfo res *verf.Result) bool {
	prog.res = append(prog.res, res)
	delete(prog.left, res.Pool)
	return len(prog.left) == 0
}
```

Reply:
Thanks for pointing it out. While using the code to verify programs I realised there are some improvements I need to make so this PR will be followed by one that increases the test coverage and adds some improvements (I don't want to clutter this PR though).